source/fuzz/fuzzer_pass_add_loads.cpp

```diff
@@ -91,35 +91,40 @@ void FuzzerPassAddLoads::Apply() {
         uint32_t memory_scope_id = 0;
         uint32_t memory_semtantics_id = 0;
 
-        // has work group or storage buffer (storage class)
-        if (chosen_instruction->GetInOperand(0).words[0] ==
-                SpvMemorySemanticsUniformMemoryMask ||
-            chosen_instruction->GetInOperand(0).words[0] ==
-                SpvMemorySemanticsWorkgroupMemoryMask) {
-          if (GetFuzzerContext()->ChoosePercentage(
-                  GetFuzzerContext()->GetChanceOfAddingAtomicLoad())) {
-            is_atomic_load = true;
-
-            memory_scope_id = FindOrCreateConstant(
-                {SpvScopeInvocation},
-                FindOrCreateIntegerType(32, GetFuzzerContext()->ChooseEven()),
-                false);
-
-            memory_semtantics_id = FindOrCreateConstant(
-                {static_cast<uint32_t>(
-                    chosen_instruction->GetInOperand(0).words[0] ==
-                            SpvMemorySemanticsUniformMemoryMask
-                        ? SpvMemorySemanticsUniformMemoryMask
-                        : SpvMemorySemanticsUniformMemoryMask)},
-                FindOrCreateIntegerType(32, GetFuzzerContext()->ChooseEven()),
-                false);
+        if (chosen_instruction->opcode() == SpvOpAccessChain) {
```

Comment:
I am afraid this is not what I meant. We want to support any op code. You do not need to check if the op code is OpAccessChain. My point was simply: your code needs to support both OpVariable and OpAccessChain (and maybe other instructions). Your previous code did not support OpAccessChain. I think your code below looks good because it looks like it supports OpVariable and OpAccessChain.

Reply:
Done.
I hope that's good now.